src/LibGPS.lua

```diff
@@ -513,10 +507,7 @@ function lib:CalculateMapMeasurements(returnToInitialMap)
         lib:PushCurrentMap()
     end
 
-    local hasWaypoint = LMP:HasMapPing(MAP_PIN_TYPE_PLAYER_WAYPOINT)
-    if(hasWaypoint) then StoreCurrentWaypoint() end

```

Comment:
I originally moved the waypoint handling out of the measurement to separate different concerns, because (re)storing the waypoint and measuring the map are two independent actions (or at least should be). Not really a fan of it, but with the need for transforming the normalized coordinates, I can see why you'd move it back in. Maybe something we can revisit in case we get an api function to get the waypoint world coordinates directly.

Reply:
Yes. Absolutely.